src/Plugin/resource/CachedResource.php

```diff
@@ -191,14 +206,24 @@ public function getControllers() {
    * {@inheritdoc}
    */
   public function index($path) {
-    return $this->subject->index($path);
+    return $this->getDataProvider()->index();
```

Comment:
Once we also have this ``$this->getPrimaryResource()`` I think we can remove all the other proxy methods, so it's always clear it's the primary resource responsible for the data. (example of such methods):

```php
  /**
   * {@inheritdoc}
   */
  public function getContext($identifier) {
    return $this->subject->getContext($identifier);
  }
```